### PR TITLE
Undeprecated the simple "fighters <name>" syntax

### PR DESCRIPTION
**Pull request by quyykk**
## Details

Undeprecates the simple fighter name syntax that got deprecated in #9029:

```
fleet "some fleet"
    "fighters" <name> # no longer deprecated
```

The reason being is that using the longer form is unnecessary if you don't need your fighters to have a different personality than the parent. ~~I've also added a warning if somebody tries to use both syntax types together.~~

## Testing Done

Verified that the deprecated warning isn't showing up anymore.

### CODE REVIEW

File: source/Fleet.cpp [hunk 1]
```diff
@@ -90,8 +90,8 @@ void Fleet::Load(const DataNode &node)
 		else if(key == "fighters" && hasValue)
 		{
 			fighterNames = GameData::Phrases().Get(child.Token(1));
-			child.PrintTrace("Warning: Deprecated use of \"fighters\" <names>."
-				" Use \"names\" <names> node under \"fighters\" instead.");
+			if(child.HasChildren())
+				child.PrintTrace("Warning: Skipping unused children of \"fighters\" tag:");
```
Comment (Hurleveur) [inline source/Fleet.cpp:94]: I agree it's more elegant this way but shouldnt we rather take the child? (bc there being a child seems more intended than just a typo there may be here?)
not really important though, just thought I'd mention it

Comment (quyykk) [reply source/Fleet.cpp:94]: idk

```
fighters "names"
    personality
        timid
```

seemed weird to me? idk

Comment (Hurleveur) [reply source/Fleet.cpp:94]: I mostly meant that
```
fighters s
    personality
        timid
```
is more likely to be a mistake (and therefore the s should be ignored)
whereas
```
fighters "fighters names"
       p
```
feels more unlikely as an error to me

but yeah I'd just parse the personality or new fighter name anyway if there is one, regrouping the two ifs into one if(fighters)

but then again it works like you did it too

Comment (Amazinite) [reply source/Fleet.cpp:94]: We should accept the name being in-line and the personality being a child. Now if for some reason the name is in both locations, the child name should take precedence. This is how ship bay hardpoints function; originally, the zoom of an engine's sprite was only defined in-line as `fighter <x> <y> <zoom>`, but even though we now have a `zoom` child of `bay`, the in-line syntax is still accepted, while the child syntax takes precedence if both exist.

**Comment by Hurleveur** [comment]
Shouldnt we have a small warning that one of the two is ignored? Idk how it goes with engines

**Comment by quyykk** [comment]
Engines don't have a warning either, that's why I didn't put one in.

File: source/Fleet.cpp [hunk 2]
```diff
@@ -87,14 +87,10 @@ void Fleet::Load(const DataNode &node)
...
-			fighterNames = GameData::Phrases().Get(child.Token(1));
-			child.PrintTrace("Warning: Deprecated use of \"fighters\" <names>."
-				" Use \"names\" <names> node under \"fighters\" instead.");
-		}
-		else if(key == "fighters" && child.HasChildren())
+		else if(key == "fighters" && (hasValue || child.HasChildren()))
 		{
+			if(hasValue)
+				fighterNames = GameData::Phrases().Get(child.Token(1));
 			for(const DataNode &grand : child)
```
Comment (Hurleveur) [inline source/Fleet.cpp:94]: it's not an issue if we dont have children and we do this, right?

Comment (Amazinite) [reply source/Fleet.cpp:94]: begin() == end() in the case of an empty list, which means the for each loop never runs. (For each loops are basically just shorthand for `for(auto it = list.begin(); it != list.end(); ++it)`.)

**Review by Hurleveur — APPROVED** [review]
looks good code wise

**Review by RisingLeaf — APPROVED** [review]

**Pull request merged**